src/components/MoneyRequestAmountInput.tsx

```diff
@@ -206,10 +220,20 @@ function MoneyRequestAmountInput(
     }));
 
     useEffect(() => {
+        const frontendAmount = onFormatAmount(amount, currency);
```

Comment:
Tests well!
We could simplify this, `setCurrentAmount(frontendAmount)` line is called twice (223 and 237), and `setSelection` is also used twice

Reply:
@kaushiktd, bump on this

Reply:
@eVoloshchak, the selection in shouldResetAmount needs to place the cursor at the end when the amount is reset, and it should also place the cursor at the end when frontendAmount is not equal to currentAmount."